sklearn/model_selection/_search.py

```diff
@@ -567,18 +572,38 @@ def fit(self, X, y=None, groups=None):
         base_estimator = clone(self.estimator)
         pre_dispatch = self.pre_dispatch
 
-        out = Parallel(
-            n_jobs=self.n_jobs, verbose=self.verbose,
-            pre_dispatch=pre_dispatch
-        )(delayed(_fit_and_score)(clone(base_estimator), X, y, self.scorer_,
-                                  train, test, self.verbose, parameters,
-                                  fit_params=self.fit_params,
-                                  return_train_score=self.return_train_score,
-                                  return_n_test_samples=True,
-                                  return_times=True, return_parameters=False,
-                                  error_score=self.error_score)
-          for train, test in cv.split(X, y, groups)
-          for parameters in candidate_params)
+        parallel = Parallel(n_jobs=self.n_jobs, verbose=self.verbose,
+                            pre_dispatch=pre_dispatch)
+
+        fit_and_score_kw = dict(
+            X=X, y=y, scorer=self.scorer_, verbose=self.verbose,
+            fit_params=self.fit_params,
+            return_train_score=self.return_train_score,
+            error_score=self.error_score,
+            return_n_test_samples=True, return_times=True,
+            return_parameters=False)
+
+        if not self.use_warm_start:
+            out = parallel(delayed(_fit_and_score)(clone(base_estimator),
+                                                   train=train, test=test,
+                                                   parameters=parameters,
+                                                   **fit_and_score_kw)
+                           for train, test in cv.split(X, y, groups)
+                           for parameters in candidate_params)
+        else:
+            # Enable warm_start on all constituent estimators
+            # XXX: is this the right thing to do?
+            base_estimator.set_params(
+                **{k: True
+                   for k in base_estimator.get_params(deep=True)
+                   if k == 'warm_start' or k.endswith('__warm_start')})
+            # one clone per fold
+            out = parallel(delayed(_warm_fit_and_score)(candidate_params,
+                                                        clone(base_estimator),
```

Comment:
clone is per fold, not per parameter candidate. all params are passed to `_warm_fit_and_score` which fits each in turn.